core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -299,18 +336,21 @@ class ZooKeeperClient(connectString: String,
     initialize()
   }
 
-  private object ZooKeeperClientWatcher extends Watcher {
+  // package level visibility for testing only
+  private[zookeeper] object ZooKeeperClientWatcher extends Watcher {
     override def process(event: WatchedEvent): Unit = {
       debug("Received event: " + event)
       Option(event.getPath) match {
         case None =>
+          val state = event.getState
+          stateToMeterMap.get(state).foreach(_.mark())
           inLock(isConnectedOrExpiredLock) {
             isConnectedOrExpiredCondition.signalAll()
           }
-          if (event.getState == KeeperState.AuthFailed) {
+          if (state == KeeperState.AuthFailed) {
             error("Auth failed.")
             stateChangeHandlers.foreach {case (name, handler) => handler.onAuthFailure()}
-          } else if (event.getState == KeeperState.Expired) {
+          } else if (state == KeeperState.Expired) {
             inWriteLock(initializationLock) {
               info("Session expired.")
               stateChangeHandlers.foreach {case (name, handler) => handler.beforeInitializingSession()}
```

Comment:
Nit: we can use `values.foreach` to simplify this. There are a few other cases where we do something similar in this method and `initialize`.

Reply:
Done.